spec/util/extensions/miq-module_spec.rb

```diff
@@ -41,6 +41,22 @@
       expect(test_module.default).to eq(value)
     end
 
+    it 'will return a prior cached value if resetting the value fails' do
+      $already_run = false
+      test_class.cache_with_timeout(:flapping_method) do
+        if $already_run == false
+          $already_run = true
+          5
+        else
+          raise "OOPS"
```

Comment:
Note, there are two scenarios I want to test but don't know how to properly write.
1) a method returns in the follow pattern:
  * 5
  * raises
  * raises
  * raises
  * ...

In this case, which is described currently in the test, we want to return the previously cached value of 5.  It makes no sense to clear the cache if we can't store a new value in the cache. We'd cache a nil.

2) a method returns in the follow pattern:
  * 5
  * raises
  * 5
  * raises
  * 5

In this case, we want to return only the cached value of 5 and never cache the nil.